routing/RouteSolver.py

```diff
@@ -0,0 +1,129 @@
+            number_of_nodes = int(input("How many nodes are in your graph? "))
+            break
+        except ValueError:
+            print('Invalid input. Please input an integer')
+            continue
+    node_count = 0
+    while node_count < int(number_of_nodes):
+        new_name = input("Please enter a name for node {node_count}\n")
+        graph.add_vertex(new_name)
+        node_count += 1
+    print(graph.vertices())
+    for node in graph.vertices():
+        another_node = True
+        while another_node:
+            node_name = input("Please enter connected node name for node {node}:\n")
+            if node_name in graph.vertices():
+                while True:
+                    direction = input("Please enter the cardinal direction traveling from {node} to {node_name}: {DIRECTIONS}\n")
+                    if direction in DIRECTIONS:
+                        graph.add_edge((node, node_name))
+                        another_connection = bool_input("Is there another connected node?")
+                        if another_connection:
+                            another_node = True
+                        else :
+                            another_node = False
+                        break
+                    else:
+                        print('Invalid direction given. Valid directions ({DIRECTIONS})')
+            else:
+                connected = bool_input("Invalid node name given. Is there a node connected?")
+                if connected:
+                    another_node = True
+                else:
+                    another_node = False
+
+def main(args):
+    if args.c:
+        load_config(args.c)
+    else:
+        get_graph_details_from_user()
+    while True:
+        start_node = input("Please enter the start node:\n")
+        if start_node in graph.vertices() :
+            break;
+        else:
+            print("Invalid start node provided\n")
+    while True:
+        end_node = input("Please enter the end node:\n")
+        if end_node in graph.vertices():
+            break;
+        else:
+            print("Invalid end node provided\n")
+    paths = graph.find_path(start_node, end_node)
+    print("This is a single path: ")
+    print(paths)
+    all_paths = graph.find_all_paths(start_node, end_node)
+    print("These are all possible paths: ")
+    print(all_paths)
+    for p in all_paths:
+        print(p)
+
+##################################################################################
+##                                                                              ##
+##                            Exeutable code below                              ##
+##                                                                              ##
+##################################################################################
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", help = "Optional config file to streamline graph creation.")
```

Comment:
add `--config` as option. No spacing between `help="Optional..."`

Reply:
-c is already an option. This is unnecessary.

Reply:
I forgot argparse does not use the term option and so I was confusing option (optional) with argument (required). It [assumes](https://stackoverflow.com/a/24181138/3990806) any argument prefixed with `-` is optional. What I meant by adding `config` as an option is the following.
```python
parser.add_argument('-c', '--config', help='...')
```
That way you could use either one depending on what level of verbosity you like. This I believe should map to
```python
args.config
```

Reply:
Also not exactly best practice to use variables that are simply `c` or `a` or `i`. Not very explicit. Requires extra time to determine what they actually mean. Should seek to avoid this and have variable/function names that are self documenting.

Reply:
I agree, however being an argument, I specified the help parameter so if a user was confused they can simply enter --h